src/main/java/bio/terra/service/dataset/flight/upgrade/predictableFileIds/ConvertToPredictableFileIdsGetIdsStep.java

```diff
@@ -0,0 +1,75 @@
+package bio.terra.service.dataset.flight.upgrade.predictableFileIds;
+
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.DatasetService;
+import bio.terra.service.filedata.FSFile;
+import bio.terra.service.filedata.FileIdService;
+import bio.terra.service.filedata.google.firestore.FireStoreDao;
+import bio.terra.service.filedata.google.firestore.FireStoreUtils;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.apache.commons.collections4.ListUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ConvertToPredictableFileIdsGetIdsStep implements Step {
+  private static final Logger logger =
+      LoggerFactory.getLogger(ConvertToPredictableFileIdsGetIdsStep.class);
+
+  private final UUID datasetId;
+  private final DatasetService datasetService;
+  private final FireStoreDao fileDao;
+  private final FileIdService fileIdService;
+
+  public ConvertToPredictableFileIdsGetIdsStep(
+      UUID datasetId,
+      DatasetService datasetService,
+      FireStoreDao fileDao,
+      FileIdService fileIdService) {
+    this.datasetId = datasetId;
+    this.datasetService = datasetService;
+    this.fileDao = fileDao;
+    this.fileIdService = fileIdService;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    Dataset dataset = datasetService.retrieve(datasetId);
+
+    // Retrieve existing file ids
+    List<String> existingFileIds = fileDao.retrieveAllFileIds(dataset, false);
+    List<FSFile> fsFiles = new ArrayList<>(existingFileIds.size());
+    for (var fileIdBatch :
+        ListUtils.partition(existingFileIds, FireStoreUtils.MAX_FIRESTORE_BATCH_SIZE)) {
+      fsFiles.addAll(fileDao.batchRetrieveById(dataset, fileIdBatch, 0));
+    }
```

Comment:
This can be done with streaming by using `flatMap()`.
```suggestion
    List<FSFile> fsFiles = ListUtils.partition(existingFileIds, FireStoreUtils.MAX_FIRESTORE_BATCH_SIZE)
        .stream()
        .flatMap(fileIdBatch -> fileDao.batchRetrieveById(dataset, fileIdBatch, 0).stream())
        .toList();
```